api/clients/v2/accountant.go

```diff
@@ -89,30 +94,39 @@ func (a *Accountant) BlobPaymentInfo(ctx context.Context, numSymbols uint64, quo
 	overflowPeriodRecord := a.GetRelativePeriodRecord(currentReservationPeriod + 2)
 	// Allow one overflow when the overflow bin is empty, the current usage and new length are both less than the limit
 	if overflowPeriodRecord.Usage == 0 && relativePeriodRecord.Usage-symbolUsage < binLimit && symbolUsage <= binLimit {
-		overflowPeriodRecord.Usage += relativePeriodRecord.Usage - binLimit
 		if err := QuorumCheck(quorumNumbers, a.reservation.QuorumNumbers); err != nil {
 			return big.NewInt(0), err
 		}
+		overflowPeriodRecord.Usage += relativePeriodRecord.Usage - binLimit
 		return big.NewInt(0), nil
 	}
 
 	// reservation not available, rollback reservation records, attempt on-demand
 	//todo: rollback on-demand if disperser respond with some type of rejection?
 	relativePeriodRecord.Usage -= symbolUsage
 	incrementRequired := big.NewInt(int64(a.PaymentCharged(numSymbols)))
-	a.cumulativePayment.Add(a.cumulativePayment, incrementRequired)
-	if a.cumulativePayment.Cmp(a.onDemand.CumulativePayment) <= 0 {
+
+	resultingPayment := big.NewInt(0)
+	resultingPayment.Add(a.cumulativePayment, incrementRequired)
+	if resultingPayment.Cmp(a.onDemand.CumulativePayment) <= 0 {
 		if err := QuorumCheck(quorumNumbers, requiredQuorums); err != nil {
 			return big.NewInt(0), err
 		}
+		a.cumulativePayment.Add(a.cumulativePayment, incrementRequired)
 		return a.cumulativePayment, nil
 	}
+	a.cumulativePayment.Add(a.cumulativePayment, incrementRequired)
```

Comment:
Ohh wait, second thought. Could we remove increment in this case? I changed the behavior on the server side that if the payment is exceeding on-chain deposit, the server will not meter the request

Reply:
removed